rasa/nlu/extractors/crf_entity_extractor.py

```diff
@@ -155,247 +158,24 @@ def train(
-            return self._from_crf_to_json(message, ents)
-        else:
-            return []
-
-    def most_likely_entity(self, idx: int, entities: List[Any]) -> Tuple[Text, Any]:
-        if len(entities) > idx:
-            entity_probs = entities[idx]
-        else:
-            entity_probs = None
-        if entity_probs:
-            label = max(entity_probs, key=lambda key: entity_probs[key])
-            if self.component_config["BILOU_flag"]:
-                # if we are using bilou flags, we will combine the prob
-                # of the B, I, L and U tags for an entity (so if we have a
-                # score of 60% for `B-address` and 40% and 30%
-                # for `I-address`, we will return 70%)
-                return (
-                    label,
-                    sum([v for k, v in entity_probs.items() if k[2:] == label[2:]]),
-                )
-            else:
-                return label, entity_probs[label]
-        else:
-            return "", 0.0
-
-    def _create_entity_dict(
-        self,
-        message: Message,
-        tokens: Union["Doc", List[Token]],
-        start: int,
-        end: int,
-        entity: str,
-        confidence: float,
-    ) -> Dict[Text, Any]:
-        if isinstance(tokens, list):  # tokens is a list of Token
-            _start = tokens[start].start
-            _end = tokens[end].end
-            value = tokens[start].text
-            value += "".join(
-                [
-                    message.text[tokens[i - 1].end : tokens[i].start] + tokens[i].text
-                    for i in range(start + 1, end + 1)
-                ]
-            )
-        else:  # tokens is a Doc
-            _start = tokens[start].idx
-            _end = tokens[start : end + 1].end_char
-            value = tokens[start : end + 1].text
-
-        return {
-            "start": _start,
-            "end": _end,
-            "value": value,
-            "entity": entity,
-            "confidence": confidence,
-        }
-
-    @staticmethod
-    def _tokens_without_cls(message: Message) -> List[Token]:
-        # [:-1] to remove the CLS token from the list of tokens
-        return message.get(TOKENS_NAMES[TEXT_ATTRIBUTE])[:-1]
-
-    def _find_bilou_end(self, word_idx, entities) -> Any:
-        ent_word_idx = word_idx + 1
-        finished = False
+        self.featurizer.process(message, **kwargs)
 
-        # get information about the first word, tagged with `B-...`
-        label, confidence = self.most_likely_entity(word_idx, entities)
-        entity_label = bilou_utils.entity_name_from_tag(label)
+        super().process(message, **kwargs)
 
-        while not finished:
-            label, label_confidence = self.most_likely_entity(ent_word_idx, entities)
+    def persist(self, file_name: Text, model_dir: Text) -> Dict[Text, Any]:
 
-            confidence = min(confidence, label_confidence)
+        self.featurizer.persist(file_name, model_dir)
```

Comment:
should file_name be different?

Reply:
We cannot easily change it without updating the interface of `persist` and `load`. I would keep it as it is as the CRFEntityExtractor is deprecated anyway.

Reply:
I meant should we add smth like `file_name + "featurizer"`, so that they don't override themselves

Reply:
The problem is that `load` does not take the filename, but just the model directory. So, not sure how to tell the featurizer that the filename differs when loading the model.

Reply:
but we can update the meta dict, will try that